### ISSUE TITLE
Adding new metadata fields to assets

### ISSUE THREAD

**Issue by davetaz** [adaptlearning/adapt_authoring#1479]
Trying to add two new fields to the asset upload screen (source and licence) however when I try and save the asset I get "ValidationError: false, false". Any idea how to fix this?

The fork can be found here, there is only one simple set of commits:

https://github.com/davetaz/adapt_authoring/tree/assetMetadata

**Comment by taylortom** [comment]
You might have to look at the `assetmanager.postAsset` function, and add the fields to the asset data. The same goes for `assetmanager.postAsset`.

**Comment by brian-learningpool** [comment]
Yes, as @taylortom has said, you've made changes to the view and the Mongoose schema, but you still have to retrieve and explicitly set the fields prior to writing to the database.

**Comment by davetaz** [comment]
Thanks @taylortom @brian-learningpool. I'll give it a go. Love the authoring tool by the way, saves us a lot of time. I'll looking to build in some sort of automated attribution system to correctly attribute assets used when required. Need the data in the database and then will use an adapt plugin to render it somewhere.

**Comment by davetaz** [comment]
Ok I have that working and now have an attribution field. Is there any way I can autofill the attribution on a graphic component when I select an asset that has attribution? Struggling to find where the assets are processed in the code. Thanks!

**Comment by davetaz** [comment]
Also want licence and source to be added to the graphic component when it is output in course.json but that probably requires a change to the graphic-component schema as well right?

**Comment by taylortom** [comment]
@davetaz Struggling to visualise the problem right now (may need more coffee 😂 ), so may be more help if I can see/test what you have so far.

**Comment by brian-learningpool** [comment]
@davetaz, yes, that would require a change to the Graphic component.

How were you thinking of rendering this?  As a concatenation on the title attribute so it appears on hover, or below the image?

**Comment by davetaz** [comment]
I hadn't really got that far. When I did this in adapt 1 we added "_licence" "_source" and "_attribution" to every image in a graphic component then I used an additional plugin to pull them out and display them all in a tutor popup. 
For example see the links at the bottom right of http://findingstories.learndata.info/#/id/57b2c1c9e396ef981f39ea8c.
Only one that doesn't work in the credits/attribution one
